pkg/cmd/diagnostics/diagnostics_download.go

```diff
@@ -0,0 +1,51 @@
+package diagnostics
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/dcos/dcos-cli/api"
+	diagnostics "github.com/dcos/dcos-core-cli/pkg/diagnostics/v2"
+	"github.com/dcos/dcos-core-cli/pkg/pluginutil"
+	"github.com/spf13/cobra"
+)
+
+func newDiagnosticsDownloadCommand(ctx api.Context) *cobra.Command {
+	var outputPath string
+
+	cmd := &cobra.Command{
+		// the <bundle-id> is used to indicate what the expected positional arg is in the help output
+		Use:   "download <bundle-id>",
+		Short: "Download diagnostics bundle",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := pluginutil.HTTPClient("")
+			client := diagnostics.NewClient(c)
+
+			id := args[0]
+
+			if outputPath == "" {
+				outputPath = fmt.Sprintf("%s.zip", id)
+			}
+
+			outFile, err := os.Create(outputPath)
+			if err != nil {
+				return err
+			}
+			defer outFile.Close()
+
+			err = client.Download(id, outFile)
+			if err != nil {
+				outFile.Close()
+				os.Remove(outputPath)
```

Comment:
I don't think cleanup is needed here: 
- user can still call the command again and it will overwrite outputPath if it already exists
- having a partially downloaded or corrupted file leaves room for debugging stuff in case of errors, also, given that diagnostics files can be quite big I can imagine that in the future we might be able to resume a broken download, like eg. `wget --continue`
- it keeps the command code simple